validator/slashing-protection/local/standard-protection-format/import.go

```diff
@@ -0,0 +1,127 @@
+
+		// Parse and transform the signed blocks data from the JSON
+		// file into the internal Prysm representation of proposal history.
+		proposalHistory, err := parseSignedBlocks(ctx, validatorData.SignedBlocks)
+		if err != nil {
+			return errors.Wrapf(err, "could not parse signed blocks in JSON file for key %#x", pubKey)
+		}
+		proposalHistoryByPubKey[pubKey] = *proposalHistory
+	}
+
+	// We save the histories to disk as atomic operations, ensuring that this only occurs
+	// until after we successfully parse all data from the JSON file. If there is any error
+	// in parsing the JSON proposal and attesting histories, we will not reach this point.
+	if err = validatorDB.SaveProposalHistoryForPubKeysV2(ctx, proposalHistoryByPubKey); err != nil {
+		return errors.Wrap(err, "could not save proposal history from imported JSON to database")
+	}
+	if err := validatorDB.SaveAttestationHistoryForPubKeysV2(ctx, attestingHistoryByPubKey); err != nil {
+		return errors.Wrap(err, "could not save attesting history from imported JSON to database")
+	}
+	return nil
+}
+
+func parseSignedBlocks(ctx context.Context, signedBlocks []*SignedBlock) (*kv.ProposalHistoryForPubkey, error) {
+	proposals := make([]kv.Proposal, len(signedBlocks))
+	for i, proposal := range signedBlocks {
+		slot, err := uint64FromString(proposal.Slot)
+		if err != nil {
+			return nil, fmt.Errorf("%d is not a valid slot: %v", slot, err)
+		}
+		var signingRoot [32]byte
+		// Signing roots are optional in the standard JSON file.
+		if proposal.SigningRoot != "" {
+			signingRoot, err = rootFromHex(proposal.SigningRoot)
+			if err != nil {
+				return nil, fmt.Errorf("%#x is not a valid root: %v", signingRoot, err)
+			}
+		}
+		proposals[i] = kv.Proposal{
+			Slot:        slot,
+			SigningRoot: signingRoot[:],
+		}
+	}
+	return &kv.ProposalHistoryForPubkey{
+		Proposals: proposals,
+	}, nil
+}
+
+func parseSignedAttestations(ctx context.Context, atts []*SignedAttestation) (*kv.EncHistoryData, error) {
+	attestingHistory := kv.NewAttestationHistoryArray(0)
+	highestEpochWritten := uint64(0)
+	var err error
+	for _, attestation := range atts {
+		target, err := uint64FromString(attestation.TargetEpoch)
+		if err != nil {
+			return nil, fmt.Errorf("%d is not a valid epoch: %v", target, err)
+		}
+		// Keep track of the highest epoch written from the imported JSON.
+		if target > highestEpochWritten {
+			highestEpochWritten = target
+		}
+		source, err := uint64FromString(attestation.SourceEpoch)
+		if err != nil {
+			return nil, fmt.Errorf("%d is not a valid epoch: %v", source, err)
+		}
+		var signingRoot [32]byte
+		// Signing roots are optional in the standard JSON file.
+		if attestation.SigningRoot != "" {
+			signingRoot, err = rootFromHex(attestation.SigningRoot)
+			if err != nil {
+				return nil, fmt.Errorf("%#x is not a valid root: %v", signingRoot, err)
+			}
+		}
+		attestingHistory, err = attestingHistory.SetTargetData(
```

Comment:
I dont think `SetTargetData` and `SetLatestEpochWritten` belong here. From the name of the function, I'd imagine that this is pure function. Putting critical stateful logic here can get dangerous